src/store/modules/job/getters.ts

```diff
@@ -57,17 +57,15 @@ const getters: GetterTree <JobState, RootState> = {
     isMiscellaneousJobsScrollable: (state) => {
       return state.miscellaneous.list?.length > 0 && state.miscellaneous.list?.length < state.miscellaneous.total
     },
-    getMoreJobs (state){
-      let jobs = state.more.draft.list.reduce((jobs: any, draftJob: any) => {
-        jobs[draftJob.systemJobEnumId] = draftJob;
-        return jobs;
-      }, {})
-      jobs = state.more.pending.list.reduce((jobs: any, pendingJob: any) => {
-        jobs[pendingJob.systemJobEnumId] = pendingJob;
-        return jobs;
-      }, jobs)
+    getMoreJobs: (state) => (jobEnums: any, enumTypeId: string): any => {
+      const orderJobEnumIds = Object.values(jobEnums) as any;
 
-      return Object.values(jobs)
+      return Object.keys(state.cached).reduce((jobs: any, enumId: string) => {
+        if(!orderJobEnumIds.includes(enumId) && state.cached[enumId]?.enumTypeId === enumTypeId) {
```

Comment:
We should first check enumTypeId as it will have many other types of jobs and it is better to avoid includes operation for other types
```suggestion
        if(state.cached[enumId]?.enumTypeId === enumTypeId && !orderJobEnumIds.includes(enumId)) {
```

Reply:
Done sir